features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/api/DeviceConfigRestService.java

```diff
@@ -126,12 +126,11 @@ Response getDeviceConfigsByInterface(
         @QueryParam("configType") @DefaultValue("") String configType);
 
     /**
-     * Delete a single device config.
-     * @param id
+     * Delete a single or multiple device config.
+     * @param id comma separated string of ids
      */
     @DELETE
-    @Path("{id : \\d+}")
-    void deleteDeviceConfig(@PathParam("id") long id);
+    Response deleteDeviceConfigs(@QueryParam("id") String id);
```

Comment:
If we really want to do this, then JAX-RS supports query parameter lists. Just make the type a list and you can specify the parameter multiple times in the URL.

Reply:
```suggestion
    Response deleteDeviceConfigs(@QueryParam("id") List<String> id);
```

Reply:
changes it to List<String> will require changes in  code like pattern matching and other, not sure if we should do this?

Reply:
> changes it to List will require changes in code like pattern matching and other, not sure if we should do this?

Instead of comma separated, it will be like this `rest/device-config/id=102&id=203`. Think that will simplify things.

Reply:
ok, required changes done

Reply:
Have you tried making this a `List<Long>`? This would save us the manual parsing and hand-crafted error handling.

Reply:
Changed to List<Long> and Implementation changed accordingly